tools/generator/Java.Interop.Tools.Generator.ObjectModel/NamespaceMapping.cs

```diff
@@ -19,9 +19,57 @@ public NamespaceMapping (IEnumerable<GenBase> gens)
 		public void Generate (CodeGenerationOptions opt, GenerationInfo gen_info)
 		{
 			using (var sw = gen_info.OpenStream (opt.GetFileName ("__NamespaceMapping__"))) {
+				sw.WriteLine ("using System;");
+				sw.WriteLine ();
+
 				foreach (var p in mappings)
 					sw.WriteLine ("[assembly:global::Android.Runtime.NamespaceMapping (Java = \"{0}\", Managed=\"{1}\")]",
 					              p.Key, p.Value);
+
+				sw.WriteLine ();
+				sw.WriteLine ("internal class JniDelegates {");
```

Comment:
Why emit these into a `JniDelegates` type instead of leaving the delegates as "global"?

Reply:
I didn't realize delegates could exist outside of a class. TIL!